sql/hive/src/test/scala/org/apache/spark/sql/hive/MetastoreDataSourcesSuite.scala

```diff
@@ -39,10 +39,8 @@ import org.apache.spark.util.Utils
 /**
  * Tests for persisting tables created though the data sources API into the metastore.
  */
-class MetastoreDataSourcesSuite extends QueryTest with SQLTestUtils with BeforeAndAfterAll
-  with Logging {
-  override def _sqlContext: SQLContext = TestHive
-  private val sqlContext = _sqlContext
+class MetastoreDataSourcesSuite
+  extends QueryTest with TestHiveSingleton with SQLTestUtils with BeforeAndAfterAll with Logging {
```

Comment:
oops I missed this, but you can remove the `BeforeAndAfterAll` here. (You get it from `SQLTestUtils`)